File: app/src/main/java/com/maufonseca/mauriciocv/presentation/main/SearchInteractor.java
package com.maufonseca.mauriciocv.presentation.main;

import com.maufonseca.mauriciocv.infrastructure.SearchEngine;

public class SearchInteractor implements SearchInteraction{

  SearchPresentation presenter;

  public SearchInteractor(SearchPresentation presenter) {
    this.presenter = presenter;
  }

  public void search(String query) {
      presenter.onSearchResults(SearchEngine.search(query));
  }
}

File: app/src/main/kotlin/com/maufonseca/mauriciocv/presentation/main/SearchInteractor.kt
package com.maufonseca.mauriciocv.presentation.main

import com.maufonseca.mauriciocv.infrastructure.SearchEngine

class SearchInteractor(var presenter: SearchPresentation) : SearchInteraction {

  override fun search(query: String) {
    presenter.onSearchResults(SearchEngine.search(query))
  }
}
